### PR TITLE
fix get_settings backwards compatibility

### PR DESCRIPTION
**Pull request by RaynorChavez**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Bug fix

* **What is the current behavior?** (You can also link to an open issue here)
users migrating from < 2.12.0 will get an error when calling `get_settings` in 2.13.0
```
{
  "message": "[{\"loc\": [\"videoPreprocessing\"], \"msg\": \"none is not an allowed value\", \"type\": \"type_error.none.not_allowed\"}, {\"loc\": [\"audioPreprocessing\"], \"msg\": \"none is not an allowed value\", \"type\": \"type_error.none.not_allowed\"}]",
  "code": "invalid_argument",
  "type": "invalid_request",
  "link": ""
}
```

* **What is the new behavior (if this is a feature change)?**
< 2.12  indexes upgraded to 2.13 will no longer get this error

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

* **Have unit tests been run against this PR?** (Has there also been any additional testing?)
Yes

* **Related Python client changes** (link commit/PR here)

* **Related documentation changes** (link commit/PR here)

* **Other information**:

* **Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

**Review by farshidz — CHANGES_REQUESTED** [review]
Can you add a test that mocks out the Vespa index settings and returns a <=2.12 index settings? Test it through the API so it covers as much as possible

**Comment by RaynorChavez** [comment]
I've added the tests you requested @farshidz as well as the thread determination unit tests

File: tests/tensor_search/test_get_settings_backwards_compatibility.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            "schema_name": "test_schema",
+            "model": Model(name="test_model", properties=None),
+            "normalize_embeddings": True,
+            "text_preprocessing": TextPreProcessing(
+                split_length=100,
+                split_overlap=0,
+                split_method=TextSplitMethod.Word
+            ),
+            "image_preprocessing": ImagePreProcessing(),
+            "video_preprocessing": None,
```
Comment (farshidz) [inline tests/tensor_search/test_get_settings_backwards_compatibility.py]: You won't get None for <2.12. They'll be missing

File: tests/tensor_search/test_media_download_thread_count.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
+import unittest
```
Comment (farshidz) [inline tests/tensor_search/test_media_download_thread_count.py]: These don't seem relevant to the PR

File: tests/tensor_search/test_get_settings_backwards_compatibility.py (hunk 1 above)
Comment (RaynorChavez) [reply tests/tensor_search/test_get_settings_backwards_compatibility.py]: I've removed the audio and video preprocessing: None lines and confirmed that tests pass

File: tests/tensor_search/test_media_download_thread_count.py (hunk 2 above)
Comment (RaynorChavez) [reply tests/tensor_search/test_media_download_thread_count.py]: Thanks, yeah they're unrelated but I added for extra coverage. I've now deleted this file and created another PR to add it: https://github.com/marqo-ai/marqo/pull/1019

**Review by farshidz — APPROVED** [review]
